src/Relude/Monoid.hs

```diff
@@ -33,3 +55,48 @@ import Relude.Monad.Reexport (Maybe, fromMaybe)
 -- []
 maybeToMonoid :: Monoid m => Maybe m -> m
 maybeToMonoid = fromMaybe mempty
+
+#if !MIN_VERSION_base(4,12,0)
+-- | This data type witnesses the lifting of a 'Monoid' into an
+-- 'Applicative' pointwise.
+--
+-- @since 4.12.0.0
+newtype Ap f a = Ap { getAp :: f a }
+    deriving ( Alternative
+             , Applicative
+             , Enum
+             , Eq
+             , Functor
+             , Generic
+             , Generic1
+             , Monad
+             , MonadFail
+             , MonadPlus
+             , Ord
+             , Read
+             , Show
+             )
+
+-- | @since 4.12.0.0
+instance (Applicative f, Semigroup a) => Semigroup (Ap f a) where
+        (Ap x) <> (Ap y) = Ap $ liftA2 (<>) x y
+
+-- | @since 4.12.0.0
+instance (Applicative f, Semigroup a, Monoid a) => Monoid (Ap f a) where
+        mempty = Ap $ pure mempty
+        mappend = (<>)
+
+-- | @since 4.12.0.0
+instance (Applicative f, Bounded a) => Bounded (Ap f a) where
+  minBound = pure minBound
+  maxBound = pure maxBound
+
+-- | @since 4.12.0.0
```

Comment:
Should we use `relude` versions here? Does it make sense to write about `base` versions?

Reply:
Good point! I will update the docs :+1: